DOC_STYLE_GUIDE.md

```diff
@@ -0,0 +1,454 @@
+
+
+## Overview
+
+We follow [_The Chicago Manual of Style_, 17th Edition](http://chicagomanualofstyle.org/), the [_Yahoo! Style Guide_](https://www.amazon.com/Yahoo-Style-Guide-Ultimate-Sourcebook/dp/031256984X), and the [Merriam-Webster dictionary](https://www.merriam-webster.com/).
+
+This style guide contains:
+
+- Deviations from and extensions to the primary style resources
+- Clarification on frequent points of confusion
+- Terminology guidelines
+
+
+## Voice
+
+The passive voice can obscure the actor and result in weak sentences. 
+In general, we prefer the active voice. For more information on this 
+topic, consult the following resources:
+
+- [Purdue Online Writing Lab](https://owl.english.purdue.edu/owl/resource/539/02/)
+- [Grammar Girl](http://www.quickanddirtytips.com/education/grammar/active-voice-versus-passive-voice)
+
+
+## Pronouns
+
+Use the second person to refer to your reader. 
+
+- **Correct**: "You must migrate your data before upgrading." 
+- **Incorrect**: "We need to migrate our data before upgrading."
+
+Use the first person plural to refer to Project Calico.
+
+- **Correct**: "We recommend five nodes."
+- **Incorrect**: "Five nodes are recommended." 
+
+
+## Anthropomorphism
+
+Avoid personifying software. 
+
+**Correct**:
+- "Kubernetes detects that Calico is installed."
+- "The delimiter specifies where to split the string."
+
+**Incorrect**:
+- "Kubernetes sees that Calico is installed."
+- "The delimiter tells Calico where to split the string."
+
+
+## Future features
+
+Avoid documenting future features or products.
+
+
+## Graphics
+
+### Screenshots and animated GIFs
+
+We discourage including screenshots and animated GIFs of web interfaces. 
+Such graphics are difficult to maintain, take up a lot of real estate, 
+and often add negligible value. 
+
+### Bitmap formats
+
+When including a static screenshot image, use the PNG format. For animated
+sequences, use GIF.
+
+### Vector formats
+
+For diagrams and other vector illustrations, we prefer the SVG format.
+
+
+## Format of source
+
+### HTML versus Markdown
+
+We prefer Markdown for readability. 
+
+### Line wraps
+
+Wrap lines at 80 characters—except [URLs](#line-breaks).
+
+
+## Punctuation
+
+### Commas
+
+Use [serial (aka Oxford) commas](https://en.wikipedia.org/wiki/Serial_comma). 
+
+### Hyphens
+
+Use hyphens only when necessary to prevent confusion. Examples include:
+- A prefix ends in the same vowel that the word begins with: "re-entry". 
+- Compound modifiers: "You can find Calico easter eggs in read-only memory."
+
+
+## Capitalization
+
+### Proper nouns
+
+Only capitalize proper nouns. 
+
+### Headings
+
+Use [sentence case](https://www.snappysentences.com/sentence-case-v-title-case/) 
+for headings.
+
+
+## Computer interfaces 
+
+### Code in headings
+
+Do not include backticks in headings. If one or more words in the heading 
+is code, add descriptive text around it.
+
+**Correct**:
+```
+# Setting the foo parameter
```

Comment:
Couple questions:
- Would it be ok to put 'foo' in quotes to better separate it?
- For a gnarlier specific case, what would it look like for the ETCD_ENDPOINTS parameter (which is used a lot in our docs)? The answer might be - avoid putting ETCD_ENDPOINTS in title's...

Reply:
So, I actually got this idea from the Google style guide, which I liked so I lifted. https://developers.google.com/style/code-in-text Except in this guide, I see that they have capitalized the name of the parameter, which I'm not as keen on. Not crazy about the quotes, either. It is just that it is awkward in a heading to have quotes. (Also awkward to have code font.)

Which brings me to ETCD_ENDPOINTS........indeed, I would recommend avoiding this if at all possible. Hope this answers your questions? LMK

Reply:
I think it could be OK to capitalize "foo" as a way of setting it apart, which is what we see in Google, if you like that we could go that way.